src/common/event.cpp

```diff
@@ -1882,24 +1882,29 @@ bool wxEvtHandler::SearchDynamicEventTable( wxEvent& event )
     wxCHECK_MSG( m_dynamicEvents, false,
                  wxT("caller should check that we have dynamic events") );
 
+    // extend the life of the wxRecursionGuardFlag
```

Comment:
I'd like to explain this in more details, e.g. something like:
```suggestion
    // Make sure that m_dynamicEvents remains valid until the end of this function,
    // even if this object itself gets deleted by one of the event handlers called
    // from here.
    //
    // We need to use m_dynamicEvents as we use its m_flag to record whether we're
    // iterating over its entries and check it at the end, at which time this object
    // itself might not exist any more.
    //
    // Note that we can't just use a local variable instead, as this function can
    // also be called recursively, if an event handler dispatches another event while
    // processing this one.
```